src/object/object_domain.c

```diff
@@ -2245,6 +2245,7 @@ tp_is_domain_cached (TP_DOMAIN * dlist, TP_DOMAIN * transient, TP_MATCH exact, T
 
 	      match = ((domain->precision == transient->precision) && (domain->collation_id == transient->collation_id)
 		       && (domain->is_desc == transient->is_desc)
+		       && (domain->codeset == transient->codeset)
```

Comment:
Is it only a problem if DB_TYPE_VARCHAR?
In the case of DB_TYPE_CHAR, is there no need to check the code set?

Reply:
Due to the added code, the assert statement at line 2279 loses its meaning.
assert (domain->codeset == transient->codeset);

Reply:
OK, good point. I fixed it for CHAR and NCHAR.